src/util/parseMarkedText.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react'
+import { Text } from 'react-native'
+
+import { styled } from '../components/hoc/styled'
+
+/**
+ * Converts formatted text into ReactNodes. The currently supported formatting
+ * rule-set is based on markdown:
+ *
+ * * Use asterisks around text to emphasize the text: \*emphasized\*
+ *
+ * @param str Formatted text with special formatting rules.
+ * @returns An array of ReactNodes which can be inserted directly into JSX
+ */
+export function parseMarkedText(str: string): React.ReactNode[] {
+  const regex = /(?<!\\)\*(.*?)(?<!\\)\*/g
```

Comment:
I see. It's "not-backlash, star, then save what comes next, not-backslash, star". Your save-what-comes-next has an extra '?' in it, since `(.*)` does the exact same thing as `(.*?)`.

This regexp also doesn't handle '\\*', but I think we can live with that for now. Writing a full parser is quite tricky, but at least this pattern covers the basic syntax anyone will ever need.

Reply:
`(.*?)` is not greedy, but `(.*)` is greedy. The `(?<!\\)` is a negative look-back for the `\*`; it says that star should not be proceeded by a backslash. We do the same thing after the non-greedy match-all. This means that we only match the non-greedy match-all group if surrounded by non-escaped stars. See the test file to prove the coverage.